File: Data Structures & Algorithms/target-sum/submission-0.cpp
class Solution {
private:
    int findWays(int target, int idx, const vector<int>& nums) {
        if (idx == nums.size()) {
            if (target == 0) {
                return 1;
            }
            return 0;
        }

        return findWays(target - nums[idx], idx + 1, nums) + findWays(target + nums[idx], idx + 1, nums);
    }
public:
    int findTargetSumWays(vector<int>& nums, int target) {
        return findWays(target, 0, nums);
    }
};

File: Data Structures & Algorithms/jump-game/submission-0.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        lower = 0
        upper = 0
        n = len(nums)
        while upper < n - 1 and upper >= lower:
            maxJump = 0
            for idx in range(lower, upper + 1):
                maxJump = max(maxJump, idx + nums[idx])
            lower = upper + 1
            upper = min(n - 1, maxJump)
        return upper == n - 1
        